godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -163,6 +163,34 @@ macro_rules! impl_vector_scalar_assign_operator {
     }
 }
 
+/// Implements a component-wise single infix binary operator between two vectors.
+macro_rules! impl_iter_vector_reduction {
+    (
+        // Name of the vector type.
+        $Vector:ty,
+        // Name of the reduction trait: `Sum` or `Product`.
+        $Operator:ident,
+        // Operator symbol: `+` or `*`.
+        $symbol:tt,
+        // Name of the function on the operator trait, for example `add`.
+        $func:ident
+    ) => {
+        impl std::iter::$Operator<Self> for $Vector {
+            fn $func<I: Iterator<Item = Self>>(iter: I) -> Self {
+                    iter.fold(<$Vector>::ZERO, |x, y| x $symbol y)
+            }
+        }
```

Comment:
Is there a reason you didn't do something more like this?
```suggestion
        impl std::iter::$Operator<Self> for $Vector {
            fn $func<I: Iterator<Item = Self>>(iter: I) -> Self {
                    Self::from_glam(iter.map(Self::to_glam).$func())
            }
        }
```

For the one below we may need to have an `as_glam(&self) -> &GlamType` impl for this same thing to work.

Reply:
I think that's a better implementation I'll test it.

Btw the weird thing is the failed unit test here works for me locally :thinking:

Reply:
the test should fail honestly, you're doing `(0,0,0,0) * (1,2,3,4) * (5,6,7,8)`, which is `(0,0,0,0)`

Reply:
Yeah, I realised afterwards haha, I think I'd run the one above in vscode.

But definitely using glam here is better than re-implementing it.

Reply:
Cool your suggestion seems to work well, thanks!

For the reference part, I just dereference it when calling `to_glam`, since it's Copy we should be fine.